src/test/java/com/wikia/webdriver/testcases/desktop/spamwikireview/ViewSpamWikisTests.java

```diff
@@ -0,0 +1,177 @@
+import com.wikia.webdriver.common.templates.NewTestTemplate;
+import com.wikia.webdriver.elements.communities.desktop.pages.SpamWikiReviewPage;
+
+import org.apache.commons.lang3.EnumUtils;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.testng.annotations.Test;
+
+@Test(groups = "SpamWikiReview")
+public class ViewSpamWikisTests extends NewTestTemplate {
+
+  /**
+   * Tests if a normal user correctly doesn't have access to Spam Wiki Review
+   */
+  @Test
+  @Execute(asUser = User.USER)
+  public void authorizationFailSpamWikiReviewTest() {
+    SpamWikiReviewPage spamWikiReviewPage = new SpamWikiReviewPage();
+    spamWikiReviewPage.open();
+
+    Assertion.assertTrue(spamWikiReviewPage.getRawVisibleBodyText()
+                             .matches(".*(User not authorized|Unauthorized).*"),
+                         "No 'unauthorized' type message was displayed to the user");
+
+    Assertion.assertTrue(spamWikiReviewPage.getListDisplayedWikisTableRows().isEmpty(),
+                         "It shouldn't be possible to see Wikis' list when unauthorized");
+  }
+
+  @Test
+  @Execute(asUser = User.STAFF)
+  public void filterJapaneseLanguageTest() {
+    SpamWikiReviewPage spamWikiReviewPage = new SpamWikiReviewPage();
+    spamWikiReviewPage.open();
+
+    // select 'ja' language
+    spamWikiReviewPage.selectLanguageOfWikis(SpamWikiReviewPage.LANGUAGE_CODE.ja);
+
+    // assert current page contains query param selecting the language
+    Assertion.assertStringContains(spamWikiReviewPage.getCurrentUrl(),
+                           SpamWikiReviewPage.LANG_QUERY_PARAM
+                           + SpamWikiReviewPage.LANGUAGE_CODE.ja);
+
+    // check if all Wikis on a presumably filtered page have the 'ja' language
+    for(WebElement wikiRow : spamWikiReviewPage.getListDisplayedWikisTableRows()) {
+      Assertion.assertEquals(wikiRow.findElement(By.xpath(SpamWikiReviewPage.XPATH_LANGUAGE_COLUMN))
+                                 .getText(), SpamWikiReviewPage.LANGUAGE_CODE.ja.toString());
+    }
+
+  }
+
+
+  @Test
+  @Execute(asUser = User.STAFF)
+  public void filterAllLanguagesTest() {
+    SpamWikiReviewPage spamWikiReviewPage = new SpamWikiReviewPage();
+    spamWikiReviewPage.open();
+
+    // select all languages
+    spamWikiReviewPage.selectLanguageOfWikis(SpamWikiReviewPage.LANGUAGE_CODE.all);
+
+
+    // assert current page contains query param selecting the language ALL
+    Assertion.assertStringContains(spamWikiReviewPage.getCurrentUrl(),
+                           SpamWikiReviewPage.LANG_QUERY_PARAM
+                           + SpamWikiReviewPage.LANGUAGE_CODE.all);
+  }
+
+
+  @Test
+  @Execute(asUser = User.STAFF)
+  public void filterOtherLanguagesTest() {
+    SpamWikiReviewPage spamWikiReviewPage = new SpamWikiReviewPage();
+    spamWikiReviewPage.open();
+
+    String spamWikiReviewListViewUrl = spamWikiReviewPage.getCurrentUrl();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "spamWikiReviewListViewUrl" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)